src/main/java/com/amazon/opendistroforelasticsearch/ad/transport/ValidateAnomalyDetectorTransportAction.java

```diff
@@ -0,0 +1,226 @@
+
+public class ValidateAnomalyDetectorTransportAction extends
+    HandledTransportAction<ValidateAnomalyDetectorRequest, ValidateAnomalyDetectorResponse> {
+    private static final Logger logger = LogManager.getLogger(ValidateAnomalyDetectorTransportAction.class);
+
+    private final Client client;
+    private final ClusterService clusterService;
+    private final NamedXContentRegistry xContentRegistry;
+    private final AnomalyDetectionIndices anomalyDetectionIndices;
+    private final SearchFeatureDao searchFeatureDao;
+    private volatile Boolean filterByEnabled;
+
+    @Inject
+    public ValidateAnomalyDetectorTransportAction(
+        Client client,
+        ClusterService clusterService,
+        NamedXContentRegistry xContentRegistry,
+        Settings settings,
+        AnomalyDetectionIndices anomalyDetectionIndices,
+        ActionFilters actionFilters,
+        TransportService transportService,
+        SearchFeatureDao searchFeatureDao
+    ) {
+        super(ValidateAnomalyDetectorAction.NAME, transportService, actionFilters, ValidateAnomalyDetectorRequest::new);
+        this.client = client;
+        this.clusterService = clusterService;
+        this.xContentRegistry = xContentRegistry;
+        this.anomalyDetectionIndices = anomalyDetectionIndices;
+        this.filterByEnabled = AnomalyDetectorSettings.FILTER_BY_BACKEND_ROLES.get(settings);
+        clusterService.getClusterSettings().addSettingsUpdateConsumer(FILTER_BY_BACKEND_ROLES, it -> filterByEnabled = it);
+        this.searchFeatureDao = searchFeatureDao;
+    }
+
+    @Override
+    protected void doExecute(Task task, ValidateAnomalyDetectorRequest request, ActionListener<ValidateAnomalyDetectorResponse> listener) {
+        User user = getUserContext(client);
+        AnomalyDetector anomalyDetector = request.getDetector();
+        try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
+            resolveUserAndExecute(
+                user,
+                AnomalyDetector.NO_ID,
+                filterByEnabled,
+                listener,
+                () -> validateExecute(request, user, listener),
+                client,
+                clusterService,
+                xContentRegistry,
+                anomalyDetector,
+                false,
+                false
+            );
+        } catch (Exception e) {
+            logger.error(e);
+            listener.onFailure(e);
+        }
+    }
+
+    private void validateExecute(
+        ValidateAnomalyDetectorRequest request,
+        User user,
+        ActionListener<ValidateAnomalyDetectorResponse> listener
+    ) {
+        ActionListener<ValidateAnomalyDetectorResponse> validateListener = ActionListener.wrap(response -> {
+            logger.info("Result of validation process " + response);
+            // forcing response to be empty
+            listener.onResponse(new ValidateAnomalyDetectorResponse((DetectorValidationIssue) null));
```

Comment:
Because ValidateAnomalyDetectorResponse also has another constructor: ValidateAnomalyDetectorResponse(StreamInput in). Casting type to `DetectorValidationIssue` makes sure we are calling correct constructor.